### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by ceohockey60**
Update and reorganize README to make it easier to navigate and use for community visitors.

### CATEGORY

Choose one

- [ ] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ x ] Documentation

### SUMMARY

Update and reorganize the README with better navigation and headline anchors to make it easier to navigate and use for community visitors. Also updated most Superset company user with the github handle of the person who filed the initial PR to give proper recognition. 

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,10 +68,8 @@ Screenshots & Gifs
 
 <kbd><img title="Visualizations" src="https://raw.githubusercontent.com/apache/incubator-superset/master/superset-frontend/images/screenshots/visualizations.png"></kbd><br/>
 
-Apache Superset
+## Why Superset
 ---------------
```
Comment (mistercrunch) [inline README.md:72]: Let's get rid of the `----------------` line here which makes a redundant `<hr/>`

**Review by mistercrunch — CHANGES_REQUESTED** [review]
Overall LGTM, good improvements, but let's get rid of the horizontal bars.

See it rendered here in your branch: https://github.com/ceohockey60/incubator-superset/tree/patch-1

**Comment by mistercrunch** [comment]
oh and master was broken earlier today :(

rebase

File: README.md (hunk 1 above)
Comment (ceohockey60) [reply README.md]: Done! :)

**Review by mistercrunch — APPROVED** [review]

**Pull request merged**